core/object/undo_redo.cpp

```diff
@@ -314,7 +316,7 @@ void UndoRedo::commit_action(bool p_execute) {
 	_redo(p_execute); // perform action
 	committing--;
 
-	if (callback && actions.size() > 0) {
+	if (add_message && callback && actions.size() > 0) {
```

Comment:
```suggestion
	if (!merging && callback && actions.size() > 0) {
```
No need for another variable here.

Reply:
It saves the `merging` state previously, it is cleared on line 306

Reply:
Ah I missed it.